torchaudio/functional/functional.py

```diff
@@ -1631,3 +1632,43 @@ def rnnt_loss(
         return costs.sum()
 
     return costs
+
+
+def rtf_power(psd_s: Tensor, psd_n: Tensor, reference_channel: Union[int, Tensor], iterations: int = 3) -> Tensor:
+    r"""Estimate the relative transfer function (RTF) or the steering vector by the power method.
+
+    Args:
+        psd_s (Tensor): The complex-valued covariance matrix of target speech.
+            Tensor of dimension `(..., freq, channel, channel)`
+        psd_n (Tensor): The complex-valued covariance matrix of noise.
+            Tensor of dimension `(..., freq, channel, channel)`
+        reference_channel (int or Tensor): Indicate the reference channel.
+            If the dtype is ``int``, it represent the reference channel index.
+            If the dtype is ``Tensor``, the dimension is `(..., channel)`, where the ``channel`` dimension
+            is one-hot.
+        iterations (int): number of iterations in power method. (Default: ``3``)
+
+    Returns:
+        Tensor: the estimated complex-valued RTF of target speech
+        Tensor of dimension `(..., freq, channel)`
+    """
+    assert iterations > 0, "The number of iteration must be greater than 0."
+    phi = torch.linalg.solve(psd_n, psd_s)  # psd_n.inv() @ psd_s
+    if torch.jit.isinstance(reference_channel, int):
+        rtf = phi[..., reference_channel]
+    elif torch.jit.isinstance(reference_channel, Tensor):
+        reference_channel = reference_channel.to(psd_n.dtype)
+        rtf = torch.einsum("...c,...c->...", [phi, reference_channel[..., None, None, :]])
+    else:
+        raise TypeError(f"Unsupported dtype for reference_channel. Found: {type(reference_channel)}.")
+    rtf = rtf.unsqueeze(-1)  # (..., freq, channel, 1)
+    if iterations >= 2:
+        # rtf = psd_s (psd_n^-1  psd_s)^(iterations - 2) psd_n^-1  psd_s  ref_channel
```

Comment:
Where is the missing 2 iterations happening?

In the `else` clause, there is one `matmul` happening and this corresponds to the number of iterations given by client code (== 1).

In the `if` clause, I see `iteration - 1` times of `matmul`. Where is the one more iteration that client code asks?

Reply:
I think `iteration=1` and `iteration=2` return the same result, which does not seem correct.

Reply:
The difference of `iteration 1` and `iteration 2` is the `psd` that rtf multiplies with:
```python
if iteration == 1:
    rtf = psd_n @ psd_n.inv() @ psd_s @ reference_channel
elif iteration == 2:
    rtf = psd_s @ psd_n.inv() @ psd_s @ reference_channel
else:
    rtf = psd_s @ (psd_n.inv() @ psd_s) ** (iteration - 2) @ psd_n.inv() @ psd_s @ reference_channel
```

Reply:
This formula can be simplified to one formula:

```
rtf = (psd_s @ psd_n.inv()) ** (iteration -1) @ psd_s @ reference_channel
```

In Pytorch, computing `psd_n.inv() @ psd_s` is easier to compute `psd_s @ psd_n.inv()`, I think that's why `phi = torch.linalg.solve(psd_n, psd_s) ` is computed first.

Reply:
ah okay, I did not see the difference of `psd_s` and `psd_n` in `torch.matmul`.

So if I understand correctly, then the initial computation of `phi` counts as one iteration, then the final `torch.matmul(psd_s, rtf)` counts as another iteration, which is why `rtf = torch.matmul(phi, rtf)` is performed `iterations - 2` times. 

If the above statement is correct, can you update the comment so that random future maintainers understand why `iterations -2`?  (and please use words instead of formula, so that SWEs without knowledge of beam forming can see and won't wonder why `iterations - 2`.

Reply:
@nateanl The existing APIs use `n_iter`. So can you use the same argument name?